website/docs/sql_dml.md

```diff
@@ -390,3 +393,72 @@ and `clean.async.enabled` options are used to disable the compaction and cleanin
+
+To utilize this feature, configure the option `index.type` as `BUCKET` and set `hoodie.index.bucket.engine` to `CONSISTENT_HASHING`.
+When enabling the consistent hashing index, it's important to enable clustering scheduling within the writer. During this process, the writer will perform dual writes for both the old and new data buckets while the clustering is pending. Although the dual write does not impact correctness, it is strongly recommended to execute clustering as quickly as possible.
+
+In the below example, we will create a datagen source and do streaming ingestion into Hudi table with consistent bucket index. In order to commit the dataset, the checkpoint needs to be enabled, here is an example configuration for a flink-conf.yaml:
+```yaml
+-- set the interval as 30 seconds
+execution.checkpointing.interval: 30000
+state.backend: rocksdb
+```
+
+```sql
+-- This is a datagen source that can generate records continuously
+CREATE TABLE sourceT (
+    uuid varchar(20),
+    name varchar(10),
+    age int,
+    ts timestamp(3),
+    `partition` as 'par1'
+) WITH (
+    'connector' = 'datagen',
+    'rows-per-second' = '200'
+);
+
+-- Create the hudi table with consistent bucket index
+CREATE TABLE t1(
+    uuid VARCHAR(20) PRIMARY KEY NOT ENFORCED,
+    name VARCHAR(10),
+    age INT,
+    ts TIMESTAMP(3),
+    `partition` VARCHAR(20)
+)
+PARTITIONED BY (`partition`)
+WITH (
+    'connector'='hudi',
+    'path' = '${work_path}/hudi-demo/hudiT',
+    'table.type' = 'MERGE_ON_READ',
+    'index.type' = 'BUCKET',
+    'clustering.schedule.enabled'='true',
+    'hoodie.index.bucket.engine'='CONSISTENT_HASHING',
+    'hoodie.clustering.plan.strategy.class'='org.apache.hudi.client.clustering.plan.strategy.FlinkConsistentBucketClusteringPlanStrategy',
+    'hoodie.clustering.execution.strategy.class'='org.apache.hudi.client.clustering.run.strategy.SparkConsistentBucketClusteringExecutionStrategy',
+    'hoodie.bucket.index.num.buckets'='8',
+    'hoodie.bucket.index.max.num.buckets'='128',
+    'hoodie.bucket.index.min.num.buckets'='8',
+    'hoodie.bucket.index.split.threshold'='1.5',
+    'write.tasks'='2'
+);
+
+-- submit the pipelines
+insert into t1 select * from sourceT;
+
+select * from t1 limit 20;
+```
+
+:::caution
+Consistent Hashing Index is supported for Flink engine since [release 0.14.0](/releases/release-0.14.0#consistent-hashing-index-support) and currently there are some limitations to use it as of 0.14.0:
+
+- This index is supported only for MOR table. This limitation also exists even if using Spark engine.
+- It does not work with metadata table enabled. This limitation also exists even if using Spark engine.
+- Consistent hashing index does not work with bulk-insert using Flink engine yet, please use simple bucket index or Spark engine for bulk-insert pipelines.
+- The resize plan which generated by Flink engine only supports merging small file groups, the file splitting is not supported yet.
```

Comment:
Typo here.

![FlinkConsistentBucketClustteringPlanStrategy](https://github.com/user-attachments/assets/8ff3bbfa-a59b-474a-b9c0-803e1ead7cc8)

Flink Consistent Hash can support split, but can't support merge.

Flink append only pipeline clustering can merge small files, but it can't be used for Consistent Hash.

Reply:
you are correct. need some improvement to the consistent hashing implementation.

Reply:
@TengHuo You're right. I would correct the doc soon. Thanks for reminding me.